### ISSUE TITLE
Issue compiling on Windows with OpenEXR 2.3 and 2.4

### ISSUE THREAD

**Issue by talonrenaud** [AcademySoftwareFoundation/OpenColorIO#894]
Since OpenEXR version 2.3 (same issue with 2.4)  I'm running into so linking issues during compilation.
I get the following error when linking:

![image](https://user-images.githubusercontent.com/6299536/69765852-6cce9d00-112a-11ea-986c-c407545d611e.png)

My current workaround is to add `#define OPENEXR_DLL` at the head of each file causing linking issues :
_(Lut1DOpData.cpp, CPUProcessor.cpp, FileFormatDiscreet1DL.cpp, MathUtils.cpp, Lut1DOpCPU.cpp, CTFTransform.cpp, ImagePacking.cpp, Lut1DOpCPU.cpp)_
or
add `OPENEXR_DLL` in Visual Studio's "C/C++ / Preprocessor / Preprocessor Definitions" settings.

I'm using CMake for both OpenEXR and OCIO compiling on Windows 10 with Visual Studio 2017, I find online people having similar issue for OpenVDB and 
https://github.com/AcademySoftwareFoundation/openvdb/pull/550
https://github.com/AcademySoftwareFoundation/openvdb/issues/496
https://github.com/AcademySoftwareFoundation/openexr/issues/302

This seems to only be problematic when using dynamically compiled OpenEXR, static builds seem to work ok from what I read in the issues linked above (I haven't tested though)
The OpenEXR team completely revamped their CMake setup in 2.3 (and 2.4) and it seems to be what caused this issue.

Is this something that could be fixed in OCIO's CMake's code ?
Thanks.

**Comment by hodoulp** [comment]
Thanks @talonrenaud for using OpenColorIO and providing feedback.

The OpenColorIO library does not directly consume the OpenEXR library. For some of its command line tools, it only needs OpenImageIO  which could or not include OpenEXR. As OpenImageIO is only an optional third-party dependency, OpenColorIO cmake only expects to find the OpenImageIO includes & libraries (i.e. no compilation of OpenImageIO by the OpenColorIO cmake).

Based on the description it seems to require a fix (or an improvement) in the OpenEXR cmake project. `unresolved external symbol` is a critical problem that needs the right fix at the right location; otherwise, it could only push the problem somewhere else or hide critical issues.

In conclusion, I would recommend you to contact the OpenEXR team by creating a new defect (when compiling OCIO command line tools using OpenImageIO (DLL) & OpenEXR (DLL) the linkage issue is also present) or updating [#302](https://github.com/AcademySoftwareFoundation/openexr/issues/302) to explain the new use case.

Please let me know if you have any comment.

**Comment by hodoulp** [comment]
@talonrenaud @meshula Can I close the defect?

**Comment by talonrenaud** [comment]
Yes, please you can go ahead an close this on my end. Thank you.
